ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -346,3 +357,57 @@ func deepCloneEvent(event *abci.Event) *abci.Event {
 	copy(clone.Attributes, event.Attributes)
 	return &clone
 }
+
+// trackCreatedPoolID tracks the created pool ID.
+// If the pool ID is not found in the event attributes, it logs an error.
+// If the pool ID is found, it parses the pool ID to uint64 and tracks it.
+func (s *indexerStreamingService) trackCreatedPoolID(event abci.Event, blockHeight int64, blockTime time.Time, txHash string) {
+	// Check if the event is pool created event
+	if event.Type != poolmanagertypes.TypeEvtPoolCreated {
+		s.logger.Error("Event type is not pool created event")
+		return
```

Comment:
We should return errors here and assert specific errors in tests. Otherwise, we cannot be sure why something was not processed by our tests.

In fact, all current tests are set up for one condition (e.g. tx hash not present) but they fail to process because we erroneously propagate `gammtypes.TypeEvtTokenSwapped` event in most relevant test cases. So our tests are incorrectly configured.

Reply:
Make sense. Done!